### ISSUE TITLE
Error: TypeError: Failed to execute 'fetch' on 'Window': Illegal invocation

### ISSUE THREAD

**Issue by moghya** [philschatz/octokat.js#180]
I'm trying to use Octokat for a simple purpose, I tried examples for browser and getting this error. I understand using callbacks is intended use but in that case, every time I'm getting the undefined result.

Please help.

**Comment by mobitar** [comment]
I'm also getting this same issue using this code and octokat 0.9.0.

```
<html>
  <body>
    <script src="./bower_components/octokat/dist/octokat.js"></script>
    <script>
      var octo = new Octokat();
      octo.repos('philschatz', 'octokat.js').fetch(function(err, repo) {
        if (err) { return alert(err); }
        alert(repo.url); 
      });
    </script>
  </body>
</html>
```

**Comment by philschatz** [comment]
In `0.9` octokat switched to using the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) which may require using a [fetch polyfill](https://github.com/github/fetch).

It seems `README.md` should be updated to clarify this because it seems that Internet Explorer does not support `window.fetch(...)`

Does the `fetch` polyfill help resolve this problem?

**Comment by azu** [comment]
Same issue on Chrome 58.0.3029.110 (64-bit).

> TypeError: 'fetch' called on an object that does not implement interface Window.

Firefox get the above error.

I found a workaround:

```js
const Octokat = require("octokat");
Octokat.Fetch = (...args) => {
    return window.fetch(...args);
};
```

Should `this` of `fetch` be `window`?

Similar issues?

- [#445 fix causes uncaught promise rejection in chrome · Issue #446 · angular/zone.js](https://github.com/angular/zone.js/issues/446 "#445 fix causes uncaught promise rejection in chrome · Issue #446 · angular/zone.js")
- [Paused on promise rejection: 'Failed to execute 'fetch' on 'Window': 1 argument required, but only 0 present.' · Issue #2122 · angular/angular-cli](https://github.com/angular/angular-cli/issues/2122 "Paused on promise rejection: &#39;Failed to execute &#39;fetch&#39; on &#39;Window&#39;: 1 argument required, but only 0 present.&#39; · Issue #2122 · angular/angular-cli")

**Comment by metasong** [comment]
donwgrade to version 0.8 solved my problem!

**Comment by philschatz** [comment]
Thanks for reporting this and sorry about the late reply.

Oops 😊 ! This was missed in the tests because they use a different implementation of fetch that does not raise this error. It should be resolved in `0.9.1`.

You can also include `Octokat.Fetch = window.fetch.bind(window)` to get around the issue.

**Comment by philschatz** [comment]
Closing for now but if this problem still occurs please comment.
